### ISSUE TITLE
Peak Calling

### ISSUE THREAD

**Issue by cadyyuheng** [timoast/signac#473]
Hi Signac Team!

I'm trying to call peaks within my dataset with CallPeaks function. However, I got an error below:

```
INFO  @ Fri, 19 Feb 2021 14:39:05: #1  tags after filtering in treatment: 44147689
INFO  @ Fri, 19 Feb 2021 14:39:05: #1  Redundant rate of treatment: 0.79
INFO  @ Fri, 19 Feb 2021 14:39:05: #1 finished!
INFO  @ Fri, 19 Feb 2021 14:39:05: #2 Build Peak Model...
INFO  @ Fri, 19 Feb 2021 14:39:05: #2 Skipped...
INFO  @ Fri, 19 Feb 2021 14:39:05: #2 Use 200 as fragment length
INFO  @ Fri, 19 Feb 2021 14:39:05: #2 Sequencing ends will be shifted towards 5' by 100 bp(s)
INFO  @ Fri, 19 Feb 2021 14:39:05: #3 Call peaks...
INFO  @ Fri, 19 Feb 2021 14:39:05: #3 Pre-compute pvalue-qvalue table...
Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/MY_OUTDIR/SeuratProject_peaks.narrowPeak': No such file or directory
```

I went to my out directory and it's empty. Do you have any suggestions on what I should look into?

Also, assume I successfully finish the peak calling step, if it's possible to conduct my analysis (e.g. co-accessibility calculation) based on peaks called by MACS3 instead of the one by cellranger? 

Thanks!!

**Comment by timoast** [comment]
Can you include the full code you're running?

**Comment by cadyyuheng** [comment]
Thanks @timoast for your quick reply! Attached is my code. One thing I should probably mention is that I conduct my previous analysis on my laptop. I moved to cluster for peak calling so I first changed the path for my fragments file.

myobj<-readRDS("mydir")
myobj@assays$peaks@fragments[[1]]@path<-"mydir/fragments.tsv.gz"
peaks <- CallPeaks(
  object = myobj,
  group.by = "act.id",
  macs2.path = "/mydir/miniconda3/bin/macs3",
  outdir = "/myout_dir/Data/peak"
)

**Comment by cadyyuheng** [comment]
I just noticed I used macs3, should I use macs2 instead? would that be an issue?

**Comment by timoast** [comment]
You should use macs2, I will add support for macs3 once there is a stable release (current release is alpha).

**Comment by cadyyuheng** [comment]
@timoast ,

I did try macs2 but received exactly the same message: 

Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/MY_OUTDIR/SeuratProject_peaks.narrowPeak': No such file or directory

**Comment by timoast** [comment]
Are you sure this directory exists: "/myout_dir/Data/peak"

**Comment by cadyyuheng** [comment]
Yep, it does exist. I noticed, even though I set the output directory to be "/myout_dir/Data/peak", the function still stores something at a temporary directory "/tmp/Rtmpbpei7z/Epithelial.bed". Would that be the issue?

INFO  @ Fri, 19 Feb 2021 15:12:49:
# Command line: **callpeak -t /tmp/Rtmpbpei7z/Epithelial.bed** -g 2.7e+09 -f BED --nomodel --extsize 200 --shift -100 -n SeuratProject --outdir /home/yfy6677/Data/peak
# ARGUMENTS LIST:
# name = SeuratProject
# format = BED
# ChIP-seq file = **['/tmp/Rtmpbpei7z/Epithelial.bed']**

**Comment by timoast** [comment]
No, that's expected. There will be temporary files generated and stored in the temp directory

**Comment by timoast** [comment]
Can you show the output of `dir.exists("/myout_dir/Data/peak")`?

**Comment by cadyyuheng** [comment]
> dir.exists("/myout_dir/Data/peak")
[1] TRUE

**Comment by timoast** [comment]
I think the issue is that you don't have write permission for that directory. Can you try changing the permissions (`chmod +rw /myout_dir/Data/peak`)

**Comment by cadyyuheng** [comment]
@timoast, thanks for the suggestion. The issue is solved as I changed the permissions and ran the code as a submitted job instead of running in the interactive session. 

A follow-up question, how could I find co-accessibility on mac2 peaks instead of using cellranger peaks?

**Comment by timoast** [comment]
You can follow the Cicero vignette, using the assay containing MACS2 peaks: https://satijalab.org/signac/articles/cicero.html

**Comment by genecell** [comment]
> @timoast, thanks for the suggestion. The issue is solved as I changed the permissions and ran the code as a submitted job instead of running in the interactive session.
> 
> A follow-up question, how could I find co-accessibility on mac2 peaks instead of using cellranger peaks?

I also met the same issue in the interactive session, and think maybe we need to add `intern = TRUE` in the `system()` command in Signac's peaks.R function: https://github.com/stuart-lab/signac/blob/fa23843c1c14b5d54d2af78fb10e9f13c5a57ead/R/peaks.R.

Best,
Min

**Comment by timoast** [comment]
That will capture the output as a string, I don't think that's what we want here.

If you're having a similar issue to the issue posted here, make sure you have write permission for the output directory.

**Comment by genecell** [comment]
> That will capture the output as a string, I don't think that's what we want here.
> 
> If you're having a similar issue to the issue posted here, make sure you have write permission for the output directory.

Thank you for your prompt response!
Finally realized it is the issue with numpy version:
```
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
``` 
Now solved it by updating the numpy version to 1.22.0, and the previous version was 1.21.5.
Thanks!

**Comment by A-legac45** [comment]
Hello, 

I am encountering a problem with call peak on three multiomic data integrated by using harmony and pre analysed on their rna side with seurat. 

My object look like this : 
> MERGE_AC_HARMO_RES_0.6
An object of class Seurat 
356627 features across 879 samples within 4 assays 
Active assay: ATAC (132979 features, 0 variable features)
 3 other assays present: RNA, peaks, new
 3 dimensional reductions calculated: pca, harmony, umap


> testpeaks2 <- CallPeaks(object = MERGE_AC_HARMO_RES_0.6, macs2.path = "/Users/alegac/opt/anaconda3/envs/MACS2/bin/macs2", outdir = "/Users/alegac/Documents/", combine.peaks = FALSE, cleanup = FALSE)


INFO  @ Mon, 03 Apr 2023 16:56:55: 
# Command line: callpeak -t /Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/projet_MAITS_Linda/atac-SEQ/atac_fragments.tsv.gz /Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/ATAC-seq_mutliomic_7_11/MULTIOME/count_CastMR1KO_CastWT_chimera_1/atac_fragments.tsv.gz /Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/ATAC-seq_mutliomic_7_11/MULTIOME/count_CastMR1KO_CastWT_chimera_2/atac_fragments.tsv.gz -g 2.7e+09 -f BED --nomodel --extsize 200 --shift -100 -n first_RNA --outdir /Users/alegac/Documents/
# ARGUMENTS LIST:
# name = first_RNA
# format = BED
# ChIP-seq file = ['/Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/projet_MAITS_Linda/atac-SEQ/atac_fragments.tsv.gz', '/Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/ATAC-seq_mutliomic_7_11/MULTIOME/count_CastMR1KO_CastWT_chimera_1/atac_fragments.tsv.gz', '/Users/alegac/Library/CloudStorage/OneDrive-INSTITUTCURIE/ATAC-seq_mutliomic_7_11/MULTIOME/count_CastMR1KO_CastWT_chimera_2/atac_fragments.tsv.gz']
# control file = None
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff = 5.00e-02
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 10000 bps
# Broad region calling is off
# Paired-End mode is off
 
INFO  @ Mon, 03 Apr 2023 16:56:55: #1 read tag files... 
INFO  @ Mon, 03 Apr 2023 16:56:55: #1 read treatment tags... 
INFO  @ Mon, 03 Apr 2023 16:56:56:  1000000 
INFO  @ Mon, 03 Apr 2023 16:56:57:  2000000 
INFO  @ Mon, 03 Apr 2023 16:56:58:  3000000 
INFO  @ Mon, 03 Apr 2023 16:56:59:  4000000 
INFO  @ Mon, 03 Apr 2023 16:57:00:  5000000 
INFO  @ Mon, 03 Apr 2023 16:57:01:  6000000 
INFO  @ Mon, 03 Apr 2023 16:57:02:  7000000 
INFO  @ Mon, 03 Apr 2023 16:57:03:  8000000 
INFO  @ Mon, 03 Apr 2023 16:57:04:  9000000 
INFO  @ Mon, 03 Apr 2023 16:57:05:  10000000 
INFO  @ Mon, 03 Apr 2023 16:57:06:  11000000 
INFO  @ Mon, 03 Apr 2023 16:57:07:  12000000 
INFO  @ Mon, 03 Apr 2023 16:57:07:  13000000 
INFO  @ Mon, 03 Apr 2023 16:57:08:  14000000 
INFO  @ Mon, 03 Apr 2023 16:57:09:  15000000 
INFO  @ Mon, 03 Apr 2023 16:57:10:  16000000 
INFO  @ Mon, 03 Apr 2023 16:57:11:  17000000 
INFO  @ Mon, 03 Apr 2023 16:57:12:  18000000 
INFO  @ Mon, 03 Apr 2023 16:57:13:  19000000 
INFO  @ Mon, 03 Apr 2023 16:57:14:  20000000 
INFO  @ Mon, 03 Apr 2023 16:57:15:  21000000 
INFO  @ Mon, 03 Apr 2023 16:57:16:  22000000 
INFO  @ Mon, 03 Apr 2023 16:57:17:  23000000 
INFO  @ Mon, 03 Apr 2023 16:57:18:  24000000 
INFO  @ Mon, 03 Apr 2023 16:57:19:  25000000 
INFO  @ Mon, 03 Apr 2023 16:57:20:  26000000 
INFO  @ Mon, 03 Apr 2023 16:57:21:  27000000 
INFO  @ Mon, 03 Apr 2023 16:57:22:  28000000 
INFO  @ Mon, 03 Apr 2023 16:57:23:  29000000 
INFO  @ Mon, 03 Apr 2023 16:57:24:  30000000 
INFO  @ Mon, 03 Apr 2023 16:57:25:  31000000 
INFO  @ Mon, 03 Apr 2023 16:57:26:  32000000 
INFO  @ Mon, 03 Apr 2023 16:57:27:  33000000 
INFO  @ Mon, 03 Apr 2023 16:57:28:  34000000 
INFO  @ Mon, 03 Apr 2023 16:57:29:  35000000 
INFO  @ Mon, 03 Apr 2023 16:57:30:  36000000 
INFO  @ Mon, 03 Apr 2023 16:57:31:  37000000 
INFO  @ Mon, 03 Apr 2023 16:57:32:  38000000 
INFO  @ Mon, 03 Apr 2023 16:57:33:  39000000 
INFO  @ Mon, 03 Apr 2023 16:57:34:  40000000 
INFO  @ Mon, 03 Apr 2023 16:57:35:  41000000 
INFO  @ Mon, 03 Apr 2023 16:57:36:  42000000 
INFO  @ Mon, 03 Apr 2023 16:57:36:  43000000 
INFO  @ Mon, 03 Apr 2023 16:57:37:  44000000 
INFO  @ Mon, 03 Apr 2023 16:57:38:  45000000 
INFO  @ Mon, 03 Apr 2023 16:57:40:  1000000 
INFO  @ Mon, 03 Apr 2023 16:57:41:  2000000 
INFO  @ Mon, 03 Apr 2023 16:57:42:  3000000 
INFO  @ Mon, 03 Apr 2023 16:57:43:  4000000 
INFO  @ Mon, 03 Apr 2023 16:57:44:  5000000 
INFO  @ Mon, 03 Apr 2023 16:57:45:  6000000 
INFO  @ Mon, 03 Apr 2023 16:57:46:  7000000 
INFO  @ Mon, 03 Apr 2023 16:57:47:  8000000 
INFO  @ Mon, 03 Apr 2023 16:57:48:  9000000 
INFO  @ Mon, 03 Apr 2023 16:57:49:  10000000 
INFO  @ Mon, 03 Apr 2023 16:57:50:  11000000 
INFO  @ Mon, 03 Apr 2023 16:57:51:  12000000 
INFO  @ Mon, 03 Apr 2023 16:57:52:  13000000 
INFO  @ Mon, 03 Apr 2023 16:57:53:  14000000 
INFO  @ Mon, 03 Apr 2023 16:57:54:  15000000 
INFO  @ Mon, 03 Apr 2023 16:57:55:  16000000 
INFO  @ Mon, 03 Apr 2023 16:57:56:  17000000 
INFO  @ Mon, 03 Apr 2023 16:57:57:  18000000 
INFO  @ Mon, 03 Apr 2023 16:57:58:  19000000 
INFO  @ Mon, 03 Apr 2023 16:57:59:  20000000 
INFO  @ Mon, 03 Apr 2023 16:58:00:  21000000 
INFO  @ Mon, 03 Apr 2023 16:58:01:  22000000 
INFO  @ Mon, 03 Apr 2023 16:58:02:  23000000 
INFO  @ Mon, 03 Apr 2023 16:58:04:  1000000 
INFO  @ Mon, 03 Apr 2023 16:58:05:  2000000 
INFO  @ Mon, 03 Apr 2023 16:58:06:  3000000 
INFO  @ Mon, 03 Apr 2023 16:58:07:  4000000 
INFO  @ Mon, 03 Apr 2023 16:58:08:  5000000 
INFO  @ Mon, 03 Apr 2023 16:58:09:  6000000 
INFO  @ Mon, 03 Apr 2023 16:58:10:  7000000 
INFO  @ Mon, 03 Apr 2023 16:58:11:  8000000 
INFO  @ Mon, 03 Apr 2023 16:58:12:  9000000 
INFO  @ Mon, 03 Apr 2023 16:58:13:  10000000 
INFO  @ Mon, 03 Apr 2023 16:58:14:  11000000 
INFO  @ Mon, 03 Apr 2023 16:58:15:  12000000 
INFO  @ Mon, 03 Apr 2023 16:58:16:  13000000 
INFO  @ Mon, 03 Apr 2023 16:58:17:  14000000 
INFO  @ Mon, 03 Apr 2023 16:58:18:  15000000 
INFO  @ Mon, 03 Apr 2023 16:58:19:  16000000 
INFO  @ Mon, 03 Apr 2023 16:58:20:  17000000 
INFO  @ Mon, 03 Apr 2023 16:58:21:  18000000 
INFO  @ Mon, 03 Apr 2023 16:58:22:  19000000 
INFO  @ Mon, 03 Apr 2023 16:58:23:  20000000 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1 tag size is determined as 74 bps 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1 tag size = 74.0 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1  total tags in treatment: 89483392 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1 user defined the maximum tags... 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1  tags after filtering in treatment: 57493804 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1  Redundant rate of treatment: 0.36 
INFO  @ Mon, 03 Apr 2023 16:58:26: #1 finished! 
INFO  @ Mon, 03 Apr 2023 16:58:26: #2 Build Peak Model... 
INFO  @ Mon, 03 Apr 2023 16:58:26: #2 Skipped... 
INFO  @ Mon, 03 Apr 2023 16:58:26: #2 Use 200 as fragment length 
INFO  @ Mon, 03 Apr 2023 16:58:26: #2 Sequencing ends will be shifted towards 5' by 100 bp(s) 
INFO  @ Mon, 03 Apr 2023 16:58:26: #3 Call peaks... 
INFO  @ Mon, 03 Apr 2023 16:58:26: #3 Pre-compute pvalue-qvalue table... 
INFO  @ Mon, 03 Apr 2023 17:00:14: #3 Call peaks for each chromosome... 
INFO  @ Mon, 03 Apr 2023 17:01:33: #4 Write output xls file... /Users/alegac/Documents/first_RNA_peaks.xls 
INFO  @ Mon, 03 Apr 2023 17:01:33: #4 Write peak in narrowPeak format file... /Users/alegac/Documents/first_RNA_peaks.narrowPeak 
INFO  @ Mon, 03 Apr 2023 17:01:33: #4 Write summits bed file... /Users/alegac/Documents/first_RNA_summits.bed 
INFO  @ Mon, 03 Apr 2023 17:01:33: Done! 
Error in file(file, "rt") : invalid 'description' argument


how to solve this?

thanks
